log4j-1.2-api/src/test/java/org/apache/log4j/CallerInformationTest.java

```diff
@@ -57,9 +57,9 @@ public void testMethodLogger() throws Exception {
         logger.warn("brains~~~");
         logger.info("Itchy. Tasty.");
         final List<String> messages = app.getMessages();
-        assertEquals("Incorrect number of messages.", 5, messages.size());
+        assertThat(messages).describedAs("Incorrect number of messages.").hasSize(5);

```

Comment:
Not that I object to these but I am not sure why the new version is "better".  In this case they both effectively say "Throw an exception with this message if the message list doesn't contain 5 items and to me both are equally clear.

Reply:
In the case the test fails fails, il the new code is able to give us more data to understand what’s broken. The failure message includes the list contents in addition to expected and actual lengths.